language/stdlib/modules/LibraTimestamp.move

```diff
@@ -48,6 +48,123 @@ module LibraTimestamp {
     public fun is_genesis(): bool acquires CurrentTimeMicroseconds {
         !::exists<CurrentTimeMicroseconds>(0xA550C18) || now_microseconds() == 0
     }
+
+    // **************** SPECIFICATIONS ****************
+
+    /*
+    This module keeps a global wall clock that stores the current Unix time in microseconds.
+    It interacts with the other modules in the following ways:
+        * Genesis: to initialize the timestamp
+        * VASP: to keep track of when credentials expire
+        * LibraSystem, LibraAccount, LibraConfig: to check if the current state is the genesis state
+        * LibraBlock: to reach consensus on the global wall clock time
+        * AccountLimits: to limit the time of account limits
+        * LibraTransactionTimeout: to determine whether a transaction is still valid
+    */
+
+    /// # Module specification
+
+    spec module {
+        /// Verify all functions in this module.
+        pragma verify = true;
+        /// The association root address.
+        define root_address(): address {
+            0xA550C18
+        }
+        /// The nil account with VM privilege.
+        define null_address(): address {
+            0x0
+        }
+        // True if the association root account has a CurrentTimeMicroseconds.
+        define root_ctm_initialized(): bool {
+            exists<CurrentTimeMicroseconds>(root_address())
+        }
+        // Auxiliary function to get the association's Unix time in microseconds.
+        define assoc_unix_time(): u64 {
+            global<CurrentTimeMicroseconds>(root_address()).microseconds
+        }
+    }
+
+    // Switch documentation context back to module level.
```

Comment:
Strictly speaken, this is only needed to switch back if you had a non-module block before.